src/backend/allocator/dumb.rs

```diff
@@ -39,8 +39,7 @@ impl<A: AsRawFd + 'static> Allocator<DumbBuffer<A>> for DrmDevice<A> {
         // dumb buffers are always linear
         if modifiers
             .iter()
-            .find(|x| **x == Modifier::Invalid || **x == Modifier::Linear)
-            .is_none()
+            .all(|&x| x != Modifier::Invalid && x != Modifier::Linear)
```

Comment:
I do not think this correct. What we want is
```suggestion
            .any(|&x| x == Modifier::Invalid || x == Modifier::Linear)
```

Reply:
I think it is correct? The original `if` triggers if all entries fail the `find` check (causing the return value to be `None`), and an entry will fail the check if it is neither `Invalid` nor `Linear`. You proposal woul cause the `if` to trigger if at least one entry is `Invalid` or `Linear`.

Reply:
Well maybe my initial implementation was already wrong then.

DumbBuffers are always going to be linear, so the user needs to be able to work with either Linear or Invalid.

Reply:
I'm not sure I follow you. The original code raises an error if the modifier list does not contain `Invalid` nor `Linear`. So was it the expected behavior or should it be changed?

Reply:
> The original code raises an error if the modifier list does not contain Invalid nor Linear.

That is expected behavior. The list needs to container either of them.

But I got confused, because we are checking here for the reverse of that to return an error, I was working with the wrong assumption on that `if`.